src/storage/src/monitor/monitored_storage_metrics.rs

```diff
@@ -124,60 +136,103 @@ impl MonitoredStorageMetrics {
+            iter_item,
+            metric_level,
+            1,
+        );
 
         let opts = histogram_opts!(
             "state_store_iter_init_duration",
             "Histogram of the time spent on iterator initialization.",
             state_store_read_time_buckets.clone(),
         );
-        let iter_init_duration =
-            register_guarded_histogram_vec_with_registry!(opts, &["table_id"], registry).unwrap();
-        let iter_init_duration = RelabeledGuardedHistogramVec::with_metric_level(
+        let iter_init_duration = register_guarded_histogram_vec_with_registry!(
+            opts,
+            &["table_id", "iter_type"],
+            registry
+        )
+        .unwrap();
+        let iter_init_duration = RelabeledGuardedHistogramVec::with_metric_level_relabel_n(
             MetricLevel::Critical,
             iter_init_duration,
             metric_level,
+            1,
         );
 
         let opts = histogram_opts!(
             "state_store_iter_scan_duration",
             "Histogram of the time spent on iterator scanning.",
             state_store_read_time_buckets.clone(),
         );
-        let iter_scan_duration =
-            register_guarded_histogram_vec_with_registry!(opts, &["table_id"], registry).unwrap();
-        let iter_scan_duration = RelabeledGuardedHistogramVec::with_metric_level(
+        let iter_scan_duration = register_guarded_histogram_vec_with_registry!(
+            opts,
+            &["table_id", "iter_type"],
+            registry
+        )
+        .unwrap();
+        let iter_scan_duration = RelabeledGuardedHistogramVec::with_metric_level_relabel_n(
             MetricLevel::Critical,
             iter_scan_duration,
             metric_level,
+            1,
         );
 
-        let iter_in_process_counts = register_int_counter_vec_with_registry!(
-            "state_store_iter_in_process_counts",
-            "Total number of iter_in_process that have been issued to state store",
-            &["table_id"],
+        let iter_counts = register_guarded_int_counter_vec_with_registry!(
+            "state_store_iter_counts",
+            "Total number of iter that have been issued to state store",
+            &["table_id", "iter_type"],
             registry
         )
         .unwrap();
-        let iter_in_process_counts = RelabeledCounterVec::with_metric_level(
+        let iter_counts = RelabeledGuardedIntCounterVec::with_metric_level_relabel_n(
             MetricLevel::Debug,
-            iter_in_process_counts,
+            iter_counts,
             metric_level,
+            1,
         );
 
+        let iter_in_progress_counts = register_guarded_int_gauge_vec_with_registry!(
+            "state_store_iter_in_progress_counts",
+            "Total number of existing iter",
+            &["table_id", "iter_type"],
+            registry
+        )
+        .unwrap();
+
+        let iter_log_op_type_counts = register_guarded_int_counter_vec_with_registry!(
```

Comment:
Only the tables that call `iter_log` will create a new label on this metrics, which is expected to be few, with just mv tables that enable log store. So I think it's fine to expose the metrics in per table granularity so that we can get more insights about the log table.